File: src/main/java/org/fountainmc/api/world/block/plants/Flower.java
package org.fountainmc.api.world.block.plants;

public interface Flower extends Plant {

    FlowerType getType();

    enum FlowerType {
        POPPY,
        BLUE_ORCHARD,
        ALLIUM,
        AZURE_BLUET,
        RED_TULIP,
        ORANGE_TULIP,
        WHITE_TULIP,
        PINK_TULIP,
        OXEYE_DAISY;
    }
}

File: src/main/kotlin/org/fountainmc/api/world/block/plants/Flower.kt
package org.fountainmc.api.world.block.plants

interface Flower : Plant {

    val type: FlowerType

    enum class FlowerType {
        POPPY,
        BLUE_ORCHARD,
        ALLIUM,
        AZURE_BLUET,
        RED_TULIP,
        ORANGE_TULIP,
        WHITE_TULIP,
        PINK_TULIP,
        OXEYE_DAISY
    }
}
